report/thesis.tex

```diff
@@ -942,7 +988,17 @@ \chapter{Related problems and future work}
             startHeight(height)
         else:
             throwError("The new set of validators doesn't have enough participants")
-    ```
+           
+    // external methods and variables
+    - getCurrentValidatorSet(): retrieves the current list of validators in the system
+    
+    - getCurrentSystemState(): retrieves the current state of the validator
+    
+    - restoreState(State): restore the validator state to the state given as parameter
+    
+    - MIN_CONSENSUS_PARTICIPANTS: defines the minimum number of validators
+    allowed by the protocol
+    
 
 Although this solution could be considered valid at first, a future study should be dedicated to the complete validation and study of this algorithm.
```

Comment:
It would be interesting to mention these three future direction (detection, recovery, efficiency enhancements) briefly in your project presentations.

Reply:
I agree, I will probably have a slide dedicated to these four directions.